src/serde.ts

```diff
@@ -0,0 +1,66 @@
+import { SRPParameters } from "./parameters";
+import { SRPRoutines } from "./routines";
+import { SRPClientSessionStep1, SRPClientSessionStep2 } from "./session-client";
+import { SRPServerSessionStep1 } from "./session-server";
+
+/**
+ * Crude JSON-based deserialization that has just enough exceptions for tssrp6a
+ * classes.
+ * Needs the serialization string and the target class prototype, e.g.
+ * `SRPServerSessionStep1.prototype`.
+ */
+export function deserialize<
+  T extends
+    | SRPServerSessionStep1
+    | SRPClientSessionStep1
+    | SRPClientSessionStep2,
+>(str: string, proto: T): T {
+  const obj = JSON.parse(str, (key, value) => {
+    switch (key) {
+      case "routines":
+        value.__proto__ = SRPRoutines.prototype;
+        return value;
+      case "parameters":
+        value.__proto__ = SRPParameters.prototype;
+        return value;
+      case "":
```

Comment:
Why empty string is also here?

Reply:
the root object value comes with empty string as key

Reply:
Thanks!